### DISCUSSION TITLE
Returning response header

### DISCUSSION THREAD

**Discussion by timpamungkas** [Netflix/dgs-framework#338] [Q&A]
DGS can receive [request header](https://netflix.github.io/dgs/datafetching/#requestheader-and-requestparam), but can I send custom response headers to graphql client?

**Comment by DiskImage** [comment]
I also need to know this to handle Authentication cookies. Can someone help us out?

**Comment by paulbakker** [comment]
It's not super convenient, but it's possible:

```java
DgsWebMvcRequestData webMvcRequestData = (DgsWebMvcRequestData)DgsContext.getRequestData(env);
        WebRequest webRequest = webMvcRequestData.getWebRequest();
        ((ServletWebRequest) webRequest).getResponse().addHeader("myheader", "hello");
```

Note that this relies on the underlying Servlet API. If you're using WebFlux you'll need to use its API instead, which you would also get from the `getRequestData` like in this example.
